### DISCUSSION TITLE
Why is `♡` a Math char but not an Emoji?

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [rust-lang/regex#1117] [Q&A] [answered]
Why is `♡` a Math char but not an Emoji?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=392c54b54c6046997283bbe6824bd079

**Comment by BurntSushi** [comment, marked as answer]
`♡` corresponds to `U+2661`.

> Why is `♡` a Math char

Because of this line

```
2660..2663    ; Math # So   [4] BLACK SPADE SUIT..BLACK CLUB SUIT
```

in [DerivedCoreProperties.txt](https://www.unicode.org/Public/UCD/latest/ucd/DerivedCoreProperties.txt) provided by the Unicode consortium.

> not an Emoji?

Becuase `U+2661` isn't listed as an `Emoji` in the [emoji data](https://www.unicode.org/Public/UCD/latest/ucd/emoji/emoji-data.txt) provided by the Unicode consortium. It _is_ listed as `Extended_Pictographic` though:

```
2661..2662    ; Extended_Pictographic# E0.0   [2] (♡..♢)    WHITE HEART SUIT..WHITE DIAMOND SUIT
```

-----

Overall, this is not a question for the regex engine. It's a question for the Unicode consortium. Moreover, I suspect this is an instance of the XY problem. I note that `\p{Emoji}` does not match every possible thing that a human thinks of as an "emoji." It couldn't possibly because character classes can only match a single codepoint. An emoji, however, can be a sequence of many codepoints. From the looks of it, you appear to have been misled by ihateregex.io. That comes as no surprise to me.

The correct definition of emoji comes from [UTS#51](http://unicode.org/reports/tr51/). It is _a lot_ more complicated than, "an emoji is a finite set of codepoints." I don't know whether you can use a regex to match emoji or not.

The fundamental issue here is that `\p{Emoji}` is poorly named. It is actually a lower level implementation detail of what makes up a proper "emoji." But given its name, it _looks_ like it's the thing you want. It isn't. It needs to be combined with other stuff.

See also:

* https://github.com/rust-lang/regex/discussions/947
* https://github.com/rust-lang/regex/issues/704
* https://github.com/BurntSushi/ripgrep/discussions/1623#discussioncomment-28827

**Reply by LuckyTurtleDev** [reply]
thanks for the fast and comprehensive answer
